frontend/src/MuiForms/DateSelector.tsx

```diff
@@ -0,0 +1,60 @@
+import 'dayjs/locale/en-gb'
+
+import { Typography } from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import { DatePicker } from '@mui/x-date-pickers'
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
+import dayjs, { Dayjs } from 'dayjs'
+import { Fragment } from 'react'
+
+interface CustomTextInputProps {
+  label?: string
+  required?: boolean
+  disabled?: boolean
+  readOnly?: boolean
+  formContext?: any
+  value: string
+  onChange: (newValue: string) => void
+  InputProps?: any
+}
+
+export default function DateInput(props: CustomTextInputProps) {
+  const { onChange, value, label, formContext } = props
+
+  const theme = useTheme()
+  const handleChange = (dateInput: Dayjs | null) => {
+    if (dateInput && dateInput.isValid()) {
+      onChange(dateInput.toDate().toISOString().split('T')[0])
```

Comment:
Using `.format()` is likely clearer than this.  E.g. `dateInput.format('YYYY-MM-DD')`.